diff --git a/platform/mikro-e/contiki-mikro-e-main.c b/platform/mikro-e/contiki-mikro-e-main.c
index fdddbff..808f8e4 100755
--- a/platform/mikro-e/contiki-mikro-e-main.c
+++ b/platform/mikro-e/contiki-mikro-e-main.c
@@ -106,31 +106,6 @@ main(int argc, char **argv)
   return 0;
 }
 /*---------------------------------------------------------------------------*/
-ISR(_CHANGE_NOTICE_VECTOR)
-{
-  if(BUTTON1_CHECK_IRQ()) {
-    /* Button1 was pressed */
-    button1_isr();
-  } else if(BUTTON2_CHECK_IRQ()) {
-    /* Button2 was pressed */
-    button2_isr();
-#ifdef MOTION_CLICK
-  } else if(MOTION_SENSOR_CHECK_IRQ()) {
-    /* Motion was detected */
-    motion_sensor_isr();
-#elif PROXIMITY_CLICK
-  } else if(PROXIMITY_SENSOR_CHECK_IRQ()) {
-    /* Proximity was detected */
-    proximity_sensor_isr();
-#endif
-  }
-  else if(INTERRUPT_CHECK_IRQ() && interrupt_isr != NULL) {
-    interrupt_isr();
-  }
-}
-
-
-/*---------------------------------------------------------------------------*/
  ISR(_EXTERNAL_1_VECTOR)
  {
     ca8210_interrupt();
