diff --git a/include/letmecreate/click/air_quality.h b/include/letmecreate/click/air_quality.h
index afc5acc..b7ea74f 100644
--- a/include/letmecreate/click/air_quality.h
+++ b/include/letmecreate/click/air_quality.h
@@ -10,7 +10,6 @@
 #define __LETMECREATE_CLICK_AIR_QUALITY_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /**
  * @brief Reads measure from Air Quality Click.
@@ -19,6 +18,6 @@
  * @param measure Indicator of air pollution in range 0..65535
  * @return 0 if successful, -1 otherwise.
  */
-int LETMECREATE_CLICK_EXPORT air_quality_click_get_measure(uint8_t mikrobus_index, uint16_t *measure);
+int air_quality_click_get_measure(uint8_t mikrobus_index, uint16_t *measure);
 
 #endif
