diff --git a/src/click/alcohol.c b/src/click/alcohol.c
index 60e18d1..c39492a 100644
--- a/src/click/alcohol.c
+++ b/src/click/alcohol.c
@@ -12,8 +12,7 @@ int alcohol_click_get_measure(uint8_t mikrobus_index, uint16_t *measure)
         fprintf(stderr, "alcohol: Cannot store measurement in null variable.\n");
         return -1;
     }
-    if (mikrobus_index != MIKROBUS_1
-    &&  mikrobus_index != MIKROBUS_2) {
+    if (mikrobus_index != MIKROBUS_1) {
         fprintf(stderr, "alcohol: Invalid mikrobus index.\n");
         return -1;
     }
