diff --git a/include/letmecreate/click/color.h b/include/letmecreate/click/color.h
index 1e7fcfe..e7501d4 100644
--- a/include/letmecreate/click/color.h
+++ b/include/letmecreate/click/color.h
@@ -11,14 +11,13 @@
 #define __LETMECREATE_CLICK_COLOR_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /**
  * @brief Enables color click.
  *
  * @return 0 if successful, otherwise it returns -1.
  */
-int LETMECREATE_CLICK_EXPORT color_click_enable(void);
+int color_click_enable(void);
 
 /**
  * @brief Reads color measurement from sensor.
@@ -29,13 +28,13 @@ int LETMECREATE_CLICK_EXPORT color_click_enable(void);
  * @param[out] blue Blue light intensity
  * @return 0 if successful, otherwise it returns -1.
  */
-int LETMECREATE_CLICK_EXPORT color_click_get_color(uint16_t *clear, uint16_t *red, uint16_t *green, uint16_t *blue);
+int color_click_get_color(uint16_t *clear, uint16_t *red, uint16_t *green, uint16_t *blue);
 
 /**
  * @brief Disables color click.
  *
  * @return 0 if successful, otherwise it returns -1.
  */
-int LETMECREATE_CLICK_EXPORT color_click_disable(void);
+int color_click_disable(void);
 
 #endif
diff --git a/src/click/color.c b/src/click/color.c
index 867d7b2..246b818 100644
--- a/src/click/color.c
+++ b/src/click/color.c
@@ -1,5 +1,7 @@
 #include <stdio.h>
-#include <time.h>
+
+#include "sys/clock.h"
+
 #include <letmecreate/click/common.h>
 #include <letmecreate/click/color.h>
 #include <letmecreate/core/i2c.h>
@@ -18,13 +20,7 @@
 
 void sleep_5ms(void)
 {
-    struct timespec req = {}, rem = {
-        .tv_sec = 0,
-        .tv_nsec = 5000000
-    };
-
-    while (nanosleep(&req, &rem))
-        req = rem;
+    clock_delay_usec(5000);
 }
 
 int color_click_enable(void)
