diff --git a/include/letmecreate/click/fan.h b/include/letmecreate/click/fan.h
index bdf3326..46a5a85 100644
--- a/include/letmecreate/click/fan.h
+++ b/include/letmecreate/click/fan.h
@@ -9,7 +9,6 @@
 #define __LETMECREATE_CLICK_FAN_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /**
  * @brief Initialise Fan Click.
@@ -18,7 +17,7 @@
  *
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT fan_click_init(void);
+int fan_click_init(void);
 
 /**
  * @brief Set speed of fan.
@@ -26,6 +25,6 @@ int LETMECREATE_CLICK_EXPORT fan_click_init(void);
  * @param[in] rpm Target speed of fan in rpm, must be in range 600, 2500
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT fan_click_set_speed(uint16_t rpm);
+int fan_click_set_speed(uint16_t rpm);
 
 #endif
