diff --git a/include/letmecreate/click/ir_distance.h b/include/letmecreate/click/ir_distance.h
index e9e4533..99d02e0 100644
--- a/include/letmecreate/click/ir_distance.h
+++ b/include/letmecreate/click/ir_distance.h
@@ -9,7 +9,6 @@
 #define __LETMECREATE_CLICK_IR_DISTANCE_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /**
  * @brief Enable IR distance click.
@@ -17,7 +16,7 @@
  * @param[in] mikrobus_index Index of the mikrobus (see #MIKROBUS_INDEX)
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT ir_distance_click_enable(uint8_t mikrobus_index);
+int ir_distance_click_enable(uint8_t mikrobus_index);
 
 /**
  * @brief Get a measurement from IR distance click.
@@ -26,7 +25,7 @@ int LETMECREATE_CLICK_EXPORT ir_distance_click_enable(uint8_t mikrobus_index);
  * @param[out] measure the value provided by the IR distance click in range 0..65535
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT ir_distance_click_get_measure(uint8_t mikrobus_index, uint16_t *measure);
+int ir_distance_click_get_measure(uint8_t mikrobus_index, uint16_t *measure);
 
 /**
  * @brief Disable IR distance click.
@@ -34,6 +33,6 @@ int LETMECREATE_CLICK_EXPORT ir_distance_click_get_measure(uint8_t mikrobus_inde
  * @param[in] mikrobus_index Index of the mikrobus (see #MIKROBUS_INDEX)
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT ir_distance_click_disable(uint8_t mikrobus_index);
+int ir_distance_click_disable(uint8_t mikrobus_index);
 
 #endif
diff --git a/src/click/ir_distance.c b/src/click/ir_distance.c
index bc00dd1..5e26ad6 100644
--- a/src/click/ir_distance.c
+++ b/src/click/ir_distance.c
@@ -14,9 +14,6 @@ static int set_enable_pin(uint8_t mikrobus_index, bool enable)
     case MIKROBUS_1:
         rst_pin = MIKROBUS_1_RST;
         break;
-    case MIKROBUS_2:
-        rst_pin = MIKROBUS_2_RST;
-        break;
     default:
         fprintf(stderr, "ir_distance: Invalid mikrobus_index.\n");
         return -1;
