diff --git a/include/letmecreate/click/joystick.h b/include/letmecreate/click/joystick.h
index ace39b3..069b1b8 100644
--- a/include/letmecreate/click/joystick.h
+++ b/include/letmecreate/click/joystick.h
@@ -11,7 +11,6 @@
 #define __LETMECREATE_CLICK_JOYSTICK_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /**
  * @brief Get the x coordinate of the joystick
@@ -20,7 +19,7 @@
  * coordinate of the click (must not be null)
  * @return 0 if successful, otherwise it returns -1.
  */
-int LETMECREATE_CLICK_EXPORT joystick_click_get_x(int8_t * x);
+int joystick_click_get_x(int8_t * x);
 
 /**
  * @brief Get the y coordinate of the joystick
@@ -29,7 +28,7 @@ int LETMECREATE_CLICK_EXPORT joystick_click_get_x(int8_t * x);
  * coordinate of the click (must not be null)
  * @return 0 if successful, otherwise it returns -1.
  */
-int LETMECREATE_CLICK_EXPORT joystick_click_get_y(int8_t * y);
+int joystick_click_get_y(int8_t * y);
 
 /**
  * @brief Get the x and y coordinates of the joystick
@@ -40,7 +39,7 @@ int LETMECREATE_CLICK_EXPORT joystick_click_get_y(int8_t * y);
  * coordinate of the click (must not be null)
  * @return 0 if successful, otherwise it returns -1.
  */
-int LETMECREATE_CLICK_EXPORT joystick_click_get_position(int8_t * x, int8_t * y);
+int joystick_click_get_position(int8_t * x, int8_t * y);
 
 #endif
 
