diff --git a/include/letmecreate/click/led_matrix.h b/include/letmecreate/click/led_matrix.h
index f524e3c..886cbbc 100644
--- a/include/letmecreate/click/led_matrix.h
+++ b/include/letmecreate/click/led_matrix.h
@@ -11,7 +11,6 @@
 #define __LETMECREATE_CLICK_LED_MATRIX_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /**
  * @brief Enable 8x8 Click.
@@ -20,7 +19,7 @@
  *
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT led_matrix_click_enable(void);
+int led_matrix_click_enable(void);
 
 /**
  * @brief Set the intensity of the LED matrix.
@@ -28,7 +27,7 @@ int LETMECREATE_CLICK_EXPORT led_matrix_click_enable(void);
  * @param[in] intensity in range 0-15
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT led_matrix_click_set_intensity(uint8_t intensity);
+int led_matrix_click_set_intensity(uint8_t intensity);
 
 /**
  * @brief Switch on/off all LED's of one column.
@@ -37,7 +36,7 @@ int LETMECREATE_CLICK_EXPORT led_matrix_click_set_intensity(uint8_t intensity);
  * @param[in] data bit string to switch on/off LED's in this column
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT led_matrix_click_set_column(uint8_t column_index, uint8_t data);
+int led_matrix_click_set_column(uint8_t column_index, uint8_t data);
 
 
 /**
@@ -46,7 +45,7 @@ int LETMECREATE_CLICK_EXPORT led_matrix_click_set_column(uint8_t column_index, u
  * @param[in] number a number between 0 and 99 which will be displayed
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT led_matrix_click_display_number(uint8_t number);
+int led_matrix_click_display_number(uint8_t number);
 
 /**
  * @brief Switch on/off all LED's.
@@ -56,7 +55,7 @@ int LETMECREATE_CLICK_EXPORT led_matrix_click_display_number(uint8_t number);
  * @param[in] columns State of all LED's (must not be null)
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT led_matrix_click_set(const uint8_t *columns);
+int led_matrix_click_set(const uint8_t *columns);
 
 /**
  * @brief Disable 8x8 Click.
@@ -65,6 +64,6 @@ int LETMECREATE_CLICK_EXPORT led_matrix_click_set(const uint8_t *columns);
  *
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT led_matrix_click_disable(void);
+int led_matrix_click_disable(void);
 
 #endif
