diff --git a/include/letmecreate/click/relay2.h b/include/letmecreate/click/relay2.h
index b56e593..5c3beb4 100644
--- a/include/letmecreate/click/relay2.h
+++ b/include/letmecreate/click/relay2.h
@@ -11,7 +11,6 @@
 #define __LETMECREATE_CLICK_RELAY2_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 /** Relay Click channel index */
 enum RELAY2_CLICK_RELAY {
@@ -27,7 +26,7 @@ enum RELAY2_CLICK_RELAY {
  * @param[in] relay Index of the relay to enable (see #RELAY2_CLICK_RELAY)
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT relay2_click_enable_relay(uint8_t mikrobus_index, uint8_t relay);
+int relay2_click_enable_relay(uint8_t mikrobus_index, uint8_t relay);
 
 /**
  * @brief Disable a relay of Relay2 Click.
@@ -36,7 +35,7 @@ int LETMECREATE_CLICK_EXPORT relay2_click_enable_relay(uint8_t mikrobus_index, u
  * @param[in] relay Index of the realy to disable (see #RELAY2_CLICK_RELAY)
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT relay2_click_disable_relay(uint8_t mikrobus_index, uint8_t relay);
+int relay2_click_disable_relay(uint8_t mikrobus_index, uint8_t relay);
 
 /**
  * @brief Set state a relay of Relay2 Click.
@@ -56,6 +55,6 @@ int relay2_click_set_relay_state(uint8_t mikrobus_index, uint8_t relay, uint8_t
  * @param[out] state 0 if off, 1 if on
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT relay2_click_get_relay_state(uint8_t mikrobus_index, uint8_t relay, uint8_t *state);
+int relay2_click_get_relay_state(uint8_t mikrobus_index, uint8_t relay, uint8_t *state);
 
 #endif
diff --git a/src/click/relay2.c b/src/click/relay2.c
index 62aae09..ad02072 100644
--- a/src/click/relay2.c
+++ b/src/click/relay2.c
@@ -8,7 +8,7 @@ static int find_gpio(uint8_t mikrobus_index, uint8_t relay)
 {
     uint8_t gpio_pin = 0;
 
-    if (mikrobus_index != MIKROBUS_1 && mikrobus_index != MIKROBUS_2) {
+    if (mikrobus_index != MIKROBUS_1) {
         fprintf(stderr, "relay2: Invalid mikrobus index.\n");
         return -1;
     }
@@ -23,11 +23,6 @@ static int find_gpio(uint8_t mikrobus_index, uint8_t relay)
             gpio_pin = MIKROBUS_1_PWM;
         else
             gpio_pin = MIKROBUS_1_AN;
-    } else { /* mikrobus_index == MIKROBUS_2 */
-        if (relay == RELAY2_CLICK_RELAY_1)
-            gpio_pin = MIKROBUS_2_PWM;
-        else
-            gpio_pin = MIKROBUS_2_AN;
     }
 
     return gpio_pin;
@@ -62,7 +57,7 @@ int relay2_click_set_relay_state(uint8_t mikrobus_index, uint8_t relay, uint8_t
     return 0;
 }
 
-int LETMECREATE_CLICK_EXPORT relay2_click_get_relay_state(uint8_t mikrobus_index, uint8_t relay, uint8_t *state)
+int relay2_click_get_relay_state(uint8_t mikrobus_index, uint8_t relay, uint8_t *state)
 {
     uint8_t gpio_pin = 0;
     int ret = 0;
