diff --git a/include/letmecreate/click/rtc.h b/include/letmecreate/click/rtc.h
index 9ec03be..d01a620 100644
--- a/include/letmecreate/click/rtc.h
+++ b/include/letmecreate/click/rtc.h
@@ -11,7 +11,6 @@
 #define __LETMECREATE_CLICK_RTC_H__
 
 #include <stdint.h>
-#include <letmecreate/click/export.h>
 
 enum WEEKDAY {
     MONDAY,
@@ -55,7 +54,7 @@ struct date {
  * used to compute year as year_offset + 2bits from RTC click.
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT rtc_click_init(uint16_t year_offset);
+int rtc_click_init(uint16_t year_offset);
 
 /**
  * @brief Read current date from RTC Click.
@@ -63,7 +62,7 @@ int LETMECREATE_CLICK_EXPORT rtc_click_init(uint16_t year_offset);
  * @param[out] date
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT rtc_click_get_date(struct date *date);
+int rtc_click_get_date(struct date *date);
 
 /**
  * @brief Set current date of RTC Click.
@@ -71,6 +70,6 @@ int LETMECREATE_CLICK_EXPORT rtc_click_get_date(struct date *date);
  * @param[in] date
  * @return 0 if successful, -1 otherwise
  */
-int LETMECREATE_CLICK_EXPORT rtc_click_set_date(struct date date);
+int rtc_click_set_date(struct date date);
 
 #endif
